modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/snapshot/SnapshotService.java

```diff
@@ -47,7 +47,8 @@
     private static final ThreadLocal<Boolean> skipSnapshotableThreadLocal = new ThreadLocal<Boolean>();
 
     private final ThreadBarrier threadBarrier;
-    private ConcurrentHashMap<String, Map<String, Snapshotable>> snapshotableMap = new ConcurrentHashMap<>();
+    private ConcurrentHashMap<String, ConcurrentHashMap<String, Snapshotable>> snapshotableMap =
```

Comment:
No need to change the declaration. Let's keep it as Map and change only the implementation to ConcurrentHashMap. Applies to all other places